ui/app/services/sockets.js

```diff
@@ -0,0 +1,36 @@
+import Service from '@ember/service';
+import config from 'nomad-ui/config/environment';
+import { getOwner } from '@ember/application';
+
+export default Service.extend({
+  getTaskStateSocket(taskState, command) {
+    const mirageEnabled =
+      config['ember-cli-mirage'] && config['ember-cli-mirage'].enabled !== false;
+
+    if (mirageEnabled) {
+      return new Object({
+        messageDisplayed: false,
+
+        send(e) {
+          if (!this.messageDisplayed) {
+            this.messageDisplayed = true;
+            this.onmessage({ data: `{"stdout":{"data":"${btoa('unsupported in Mirage\n\r')}"}}` });
```

Comment:
This is a really nice touch. I'm sure this will save someone in the future some time.